src/frontends/tensorflow_common/src/op/adjust_saturation.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reduce_mean.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/maximum.hpp"
+#include "openvino/op/minimum.hpp"
+#include "openvino/op/reduce_max.hpp"
+#include "openvino/op/reduce_min.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/split.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/floor_mod.hpp"
+#include "openvino/op/floor.hpp"
+#include "openvino/op/abs.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/clamp.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+shared_ptr<tuple<shared_ptr<Node>, shared_ptr<Node>, shared_ptr<Node>>> convert_rgb_to_hsv(shared_ptr<Node> images, element::Type type) {
+    // image format conversion based on https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/adjust_saturation_op.cc
+    auto zero = make_shared<v0::Constant>(type, Shape{}, 0.0f);
+    auto one = make_shared<v0::Constant>(type, Shape{}, 1.0f);
+
+    // find max and min across channel axis: max = value (V)
+    auto max_rgb = make_shared<v1::ReduceMax>(images, make_shared<v0::Constant>(element::i64, Shape{1}, vector<int64_t>{-1}), true);
+    auto min_rgb = make_shared<v1::ReduceMin>(images, make_shared<v0::Constant>(element::i64, Shape{1}, vector<int64_t>{-1}), true);
+    auto range = make_shared<v1::Subtract>(max_rgb, min_rgb);
+    auto vv = max_rgb;
+
+    // compute Saturation (S)
+    auto ss_ = make_shared<v1::Divide>(range, vv);
+    auto ss = make_shared<v1::Select>(make_shared<v1::Greater>(vv, zero), ss_, zero);
+
+    // compute normalization factor (for Hue calculation)
+    auto norm = make_shared<v1::Divide>(
+        make_shared<v0::Constant>(type, range->get_shape(), vector<float>{1.0f}),
+        make_shared<v1::Multiply>(
+            make_shared<v0::Constant>(type, range->get_shape(), vector<float>{6.0f}),
+            range
+        )
+    );
+
+    // split the image tensor into R, G, B channels
+    int batch_dim = images->get_shape().size() - 1;
```

Comment:
I think we can use `axis` const equal to `-1` for `Split` operation. `batch_dim` is not needed and not correct for general case (dynamic dimensions and/or dynamic rank for input), please remove,